include/boost/geometry/algorithms/detail/envelope/range.hpp

```diff
@@ -63,24 +57,29 @@ struct envelope_range
         // initialize MBR
         initialize<Box, 0, dimension<Box>::value>::apply(mbr);
 
-        Iterator it = first;
         if (it != last)
         {
             // initialize box with first element in range
-            dispatch::envelope<value_type>::apply(*it, mbr, strategy);
+            dispatch::envelope
+                <
+                    value_type
+                >::apply(*it, mbr, strategy.get_element_envelope_strategy());
 
             // consider now the remaining elements in the range (if any)
             for (++it; it != last; ++it)
             {
-                dispatch::expand<Box, value_type>::apply(mbr, *it, strategy);
+                dispatch::expand
+                    <
+                        Box, value_type
+                    >::apply(mbr, *it, strategy.get_element_expand_strategy());
             }
         }
     }
 
     template <typename Range, typename Box, typename Strategy>
     static inline void apply(Range const& range, Box& mbr, Strategy const& strategy)
     {
-        return apply(boost::begin(range), boost::end(range), mbr, strategy);
+        return apply(Strategy::begin(range), Strategy::end(range), mbr, strategy);
```

Comment:
This is surprising. What is this for kind of strategy? So it should implement range like features?

Reply:
The problem with envelope is that in cartesian we can process points but in spherical and geographic we have to analyse segments becuase geodesic's vertex may have different latitude (higher or lower on northern or southern hemisphere respectively) than the endpoints. So the strategy feeds the algorithm either with points or with segments. Previously the dispatching by CS was done in the algorithm. I decided to do it this way to allow the strategy to define the smallest Cs-specific portions of the algorithm. Otherwise I'd have to implement the whole calculation of envelope of a range as strategy.

Reply:
+1